src/platform/Darwin/DnssdContexts.cpp

```diff
@@ -709,7 +712,16 @@ void ResolveContext::OnNewInterface(uint32_t interfaceId, const char * fullname,
     // resolving.
     interface.fullyQualifiedDomainName = hostnameWithDomain;
 
-    interfaces.insert(std::make_pair(interfaceId, std::move(interface)));
+    std::string domainFromHostname = GetDomainFromHostName(hostnameWithDomain);
+    if (domainFromHostname.empty())
+    {
+        ChipLogError(Discovery, "Mdns: No domain set in hostname %s", hostnameWithDomain);
+        return;
+    }
+
+    std::pair<uint32_t, std::string> interfaceKey = std::make_pair(interfaceId, domainFromHostname);
```

Comment:
This would have been a _great_ place to use `auto`.

Reply:
done.